handler.c

```diff
@@ -1,562 +1,546 @@
-	if ( stat( path, &st ) != 0 ) {
-	    fprintf( stderr, "stat %s: %s\n", path, strerror( errno ));
-	    continue;
-	}
-	if ( !S_ISREG( st.st_mode )) {
-	    continue;
-	}
-	
-	/*
-	 * build a sorted list of files in the directory
-	 * so the admin can dictate in what order they'll
-	 * be processed, allowing handler precedence.
-	 *
-	 * HFS+ appears to store files in lexical order
-	 * on disk, but we can't count on an HFS+ volume.
-	 */
-	lladd( path, &head );
-    }
-    if ( closedir( d ) != 0 ) {
-	fprintf( stderr, "closedir: %s\n", strerror( errno ));
-	rc = 1;
-    }
-
-    for ( cur = head; cur != NULL; cur = tmp ) {
-	dhandler_f = fsethandler;
-	if (( p = strrchr( cur->l_path, '.' )) != NULL ) {
-	    p++;
-	    if ( strcmp( p, "plist" ) == 0 ) {
-		dhandler_f = psethandler;
-	    }
-	}
-
-	if ( verbose ) {
-	    printf( "Applying settings from %s\n", cur->l_path );
-	}
-	if ( dhandler_f( cur->l_path ) != 0 ) {
-	    rc = 1;
-	}
-
-	tmp = cur->l_next;
-	free( cur->l_path );
-	free( cur );
-    }
-
-    return( rc );
-}
+  if (c2cf(uti, &cfuti) != 0) {
+    return (2);
+  }
 
-    int
-uti_handler_show( char *uti, int showall )
-{
-    CFArrayRef		cfhandlers = NULL;
-    CFStringRef		cfuti = NULL;
-    CFStringRef		cfhandler = NULL;
-    char		dh[ MAXPATHLEN ];
-    int			rc = 0;
-    int			count, i;
-
-    if ( uti == NULL ) {
-	fprintf( stderr, "Invalid UTI.\n" );
-	return( 2 );
-    }
-
-    if ( c2cf( uti, &cfuti ) != 0 ) {
-	return( 2 );
-    }
-
-    if ( showall ) {
-	if (( cfhandlers = LSCopyAllRoleHandlersForContentType(
-				cfuti, kLSRolesAll )) == NULL ) {
-	    if (( cfhandlers = LSCopyAllHandlersForURLScheme(
-					cfuti )) == NULL ) {
-		fprintf( stderr, "%s: no handlers\n", uti );
-		rc = 1;
-		goto uti_show_done;
-	    }
-	}
-
-	if ( verbose ) {
-	    printf( "All handlers for %s:\n", uti );
-	}
-
-	count = CFArrayGetCount( cfhandlers );
-	for ( i = 0; i < count; i++ ) {
-	    cfhandler = CFArrayGetValueAtIndex( cfhandlers, i );
-	    if ( cf2c( cfhandler, dh, sizeof( dh )) != 0 ) {
-		rc = 2;
-		continue;
-	    }
-	    printf( "%s\n", dh );
-	    memset( dh, 0, sizeof( dh ));
-	}
-	
-	cfhandler = NULL;
-    } else {
-	if (( cfhandler = LSCopyDefaultRoleHandlerForContentType(
-				cfuti, kLSRolesAll )) == NULL ) {
-	    if (( cfhandler = LSCopyDefaultHandlerForURLScheme(
-					cfuti )) == NULL ) {
-		fprintf( stderr, "%s: no default handler\n", uti );
-		rc = 1;
-		goto uti_show_done;
-	    }
-	}
-
-	if ( cf2c( cfhandler, dh, MAXPATHLEN ) != 0 ) {
-	    rc = 2;
-	    goto uti_show_done;
-	}
-
-	if ( verbose ) {
-	    printf( "Default handler for %s: ", uti );
-	}
-	printf( "%s\n", dh );
+  if (showall) {
+    if ((cfhandlers =
+             LSCopyAllRoleHandlersForContentType(cfuti, kLSRolesAll)) == NULL) {
+      if ((cfhandlers = LSCopyAllHandlersForURLScheme(cfuti)) == NULL) {
+        fprintf(stderr, "%s: no handlers\n", uti);
+        rc = 1;
+        goto uti_show_done;
```

Comment:
GOTOs are all a computer ever really does ;-) (JMP, JE, JG, etc.) But yes, this is part of the original code and I'll get it in a future code cleanup.